pyro/distributions/categorical.py

```diff
@@ -94,17 +87,13 @@ def shape(self, x=None):
         """
```

Comment:
Not introduced in this PR, but we will have to think about what the `event_shape` returns for `Categorical`. It should not be `ps.size()[:-event_dim]` which it returns now. I suppose just returning `(1,)` is best.

Reply:
Actually, we are using the event dimension to broadcast the parameters at many places, and with this we are running into this weird issue because the shape of the event dimension for the params is not the same as that of data. e.g.

```
ps = self.ps.expand(x_shape[:-event_dim] + self.event_shape()) # will not work if event_shape returns (1,)
```

So this will need a bit more work than just changing to `(1,)`. We can punt on this (`xskip` shape related tests) until we find a better solution.